allennlp/models/crf_tagger.py

```diff
@@ -0,0 +1,210 @@
+from typing import Dict, List, Optional
+
+import numpy
+from overrides import overrides
+import torch
+from torch.nn.modules.linear import Linear
+
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.data import Vocabulary
+from allennlp.modules import Seq2SeqEncoder, TimeDistributed, TextFieldEmbedder, ConditionalRandomField
+from allennlp.models.model import Model
+from allennlp.nn import InitializerApplicator, RegularizerApplicator
+from allennlp.nn.util import get_text_field_mask, viterbi_decode
+from allennlp.training.metrics import SpanBasedF1Measure
+
+START_TAG = "@@START@@"
+END_TAG = "@@END@@"
+
+@Model.register("crf_tagger")
+class CrfTagger(Model):
+    """
+    The ``CrfTagger`` encodes a sequence of text with a stacked ``Seq2SeqEncoder``,
+    then uses a Conditional Random Field model to predict a tag for each token in the sequence.
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``, required
+        A Vocabulary, required in order to compute sizes for input/output projections.
+    text_field_embedder : ``TextFieldEmbedder``, required
+        Used to embed the ``tokens`` ``TextField`` we get as input to the model.
+    stacked_encoder : ``Seq2SeqEncoder``
+        The encoder (with its own internal stacking) that we will use in between embedding tokens
+        and predicting output tags.
+    label_namespace: ``str``, optional (default=``"labels"``)
+        We'll need to add special START and END tags to whatever namespace holds our labels.
+        Unless you did something unusual, the default value should be what you want.
+    initializer : ``InitializerApplicator``, optional (default=``InitializerApplicator()``)
+        Used to initialize the model parameters.
+    regularizer : ``RegularizerApplicator``, optional (default=``None``)
+        If provided, will be used to calculate the regularization penalty during training.
+    """
+
+    def __init__(self, vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,
+                 stacked_encoder: Seq2SeqEncoder,
+                 label_namespace: str = "labels",
+                 initializer: InitializerApplicator = InitializerApplicator(),
+                 regularizer: Optional[RegularizerApplicator] = None) -> None:
+        super().__init__(vocab, regularizer)
+
+        self.label_namespace = label_namespace
+
+        self.text_field_embedder = text_field_embedder
+
+        # Make sure we have START and END tags
+        self.start_tag = vocab.add_token_to_namespace(START_TAG, label_namespace)
```

Comment:
These are the indices of the tags, not the tags themselves - maybe `self.start_tag_index`?